### PR TITLE
added check of var sizes when loading a pre-existing coloring

### PR DESCRIPTION
**Pull request by naylor-b**

### CODE REVIEW

File: openmdao/core/tests/test_coloring.py [hunk 1]
```diff
@@ -942,6 +942,193 @@ def test_multipoint_with_coloring(self):
...
+            if fixed:
+                p.driver.use_fixed_coloring()
+
+        indeps = model.add_subsystem('indeps', om.IndepVarComp())
+        for name, sz in zip(wrtnames, sizes):
+            indeps.add_output(name, val=np.ones(sz))
+            model.add_design_var('indeps.' + name)
+
+        for name, sz in zip(ofnames, sizes):
+            model.add_constraint('comp.' + name, lower=0.0)
```
Comment (swryan) [inline openmdao/core/tests/test_coloring.py]: `sz` is not used

File: openmdao/utils/coloring.py [hunk 2]
```diff
@@ -395,27 +391,75 @@ def _check_config_partial(self, system):
...
+            for i, (my_sz, sz) in enumerate(zip(self._col_var_sizes, wrt_sizes)):
+                if my_sz != sz:
+                    changed_sizes.append(wrt_names[i])
+
+        if changed_sizes:
+            msg.append('The following variables have changed sizes: %s.' % sorted(changed_sizes))
+
+        if len(msg) > 1:
+            msg.append(msg_suffix)
+            raise RuntimeError(' '.join(msg))
```
Comment (swryan) [inline openmdao/utils/coloring.py]: Would `'n'.join` be better?
Seeing the long unbroken messages in the tests, wonder if line breaks would make it more readable?

File: openmdao/core/tests/test_coloring.py (hunk 1 above)
Comment (naylor-b) [reply openmdao/core/tests/test_coloring.py]: fixed

File: openmdao/utils/coloring.py (hunk 2 above)
Comment (naylor-b) [reply openmdao/utils/coloring.py]: fixed

**Pull request merged**